specification/cdn/resource-manager/Microsoft.Cdn/preview/2019-06-15-preview/cdnwebapplicationfirewall.json

```diff
@@ -0,0 +1,919 @@
+            "Deleting"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PolicyResourceState",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "policySettings": {
+      "description": "Defines contents of a web application firewall global configuration",
+      "properties": {
+        "enabledState": {
+          "description": "describes if the policy is in enabled state or disabled state",
+          "type": "string",
+          "enum": [
+            "Disabled",
+            "Enabled"
+          ],
+          "x-ms-enum": {
+            "name": "PolicyEnabledState",
+            "modelAsString": true
+          }
+        },
+        "mode": {
+          "description": "Describes if it is in detection mode or prevention mode at policy level.",
+          "type": "string",
+          "enum": [
+            "Prevention",
+            "Detection"
+          ],
+          "x-ms-enum": {
+            "name": "PolicyMode",
+            "modelAsString": true
+          }
+        },
+        "defaultRedirectUrl": {
+          "description": "If action type is redirect, this field represents the default redirect URL for the client.",
+          "type": "string"
+        },
+        "defaultCustomBlockResponseStatusCode": {
+          "description": "If the action type is block, this field defines the default customer overridable http response status code.",
+          "type": "integer",
+          "maximum": 599,
+          "exclusiveMaximum": false,
+          "minimum": 200,
+          "exclusiveMinimum": false
+        },
+        "defaultCustomBlockResponseBody": {
+          "description": "If the action type is block, customer can override the response body. The body must be specified in base64 encoding.",
+          "type": "string",
+          "pattern": "^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{4})$"
+        }
+      }
+    },
+    "CdnEndpoint": {
+      "description": "Defines the ARM Resource ID for the linked endpoints",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "ARM Resource ID string."
+        }
+      }
+    },
+    "CustomRuleList": {
+      "description": "Defines contents of custom rules",
+      "properties": {
+        "rules": {
+          "description": "List of rules",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CustomRule"
+          }
+        }
+      }
+    },
+    "CustomRule": {
+      "description": "Defines the common attributes for a custom rule that can be included in a waf policy",
+      "type": "object",
+      "required": [
+        "name",
+        "priority",
+        "matchConditions",
+        "action"
+      ],
+      "properties": {
+        "name": {
+          "description": "Defines the name of the custom rule",
+          "type": "string"
```

Comment:
@devigned as we discussed in the Api review meeting, the name is consistent with afd's waf which is frontdoorwebapplicationfirewallpolicies, we deliberated on this internally and decided to go with CdnWebApplicationFirewallPolicies instead of cdnwafpolicies.

As far as the rule name is concerned, it is arbitrary at the moment, we will look into adding more constraints as part of the work for the "Stable" version.

Reply:
So I can provide you a 10000 char string?

Reply:
please add constraints in stable